ios/RNSScreen.m

```diff
@@ -612,6 +636,53 @@ - (void)notifyFinishTransitioning
   [RNSScreenWindowTraits updateWindowTraits];
 }
 
+# pragma mark - transition progress related methods
+
+- (void)setupProgressNotification
+{
+  if (self.transitionCoordinator != nil && !_isSendingProgress) {
+    _isSendingProgress = YES;
+    UIView *fakeView = [UIView new];
+    fakeView.alpha = 0.0;
+    _fakeView = fakeView;
+    [self.transitionCoordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+      [[context containerView] addSubview:fakeView];
+      fakeView.alpha = 1.0;
+      self->_animationTimer = [CADisplayLink displayLinkWithTarget:self selector:@selector(handleAnimation)];
+      [self->_animationTimer addToRunLoop:[NSRunLoop currentRunLoop] forMode:NSDefaultRunLoopMode];
+    } completion:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+      [self->_animationTimer setPaused:YES];
+      [self->_animationTimer invalidate];
+      self->_fakeView = nil;
```

Comment:
do we need fakeView to be retained under `this`?

Reply:
We use it in every tick of `_animationTimer` and I am not sure how else can it be used in the `handleAnimation` of it (see https://github.com/software-mansion/react-native-screens/pull/890/files#diff-e988f8ab84e7063d6cc9ad4e634806389882ce6727b5c95f3cfbc9f9d3921a67R665)